react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -1,38 +1,39 @@
+/* eslint-disable no-console */
 import React, { Component, Fragment } from 'react';
-import store from '@redux/store';
+import { connect } from 'react-redux';
 import Navbar from '@components/Navbar';
 import Footer from '@components/Footer';
+import { func, arrayOf } from 'prop-types';
+import { bookPropType } from '@constants/propTypes';
+
+import actionsCreators from '../../../redux/book/actions';
 
 import Book from './components/Book';
 import Search from './components/Search';
 import ShoppingCart from './components/ShoppingCart';
 import styles from './styles.scss';
 
 class App extends Component {
-  state = {
-    books: [],
-    bookSelected: []
-  };
+  componentDidMount = () => this.props.getBooks();
 
-  componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
-  }
-
-  // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    const books = this.props.originalData.filter(book => book.name.toLowerCase().includes(value));
+    this.props.searchBooks(books);
+  };
 
-  // TODO to implement the dispatch
-  addToCart = item => {};
+  addToCart = item => this.props.addToCart(item);
 
-  // TODO to implement the dispatch
-  addItem = itemId => {};
+  addItem = itemId => {
+    const books = [...this.props.bookSelected];
+    const book = books.find(({ id }) => id === itemId);
+    book.quantity += 1;
```

Comment:
You can do this instead 

```suggestion
    book.quantity++;
```